odbc/src/api/functions.rs

```diff
@@ -1794,12 +1794,468 @@ pub unsafe extern "C" fn SQLGetInfo(
+            i16_len::set_output_fixed_data(&SQL_CL_START, info_value_ptr, string_length_ptr)
+        }
+        // SQL_SQL_CONFORMANCE
+        InfoType::SqlConformance => {
+            // MongoSQL is SQL-92 Entry level compliant.
+            i16_len::set_output_fixed_data(
+                &SQL_SC_SQL92_ENTRY,
+                info_value_ptr,
+                string_length_ptr,
+            )
+        }
+        // SQL_ODBC_INTERFACE_CONFORMANCE
+        InfoType::OdbcInterfaceConformance => {
+            // The MongoSQL ODBC Driver currently meets the minimum compliance level.
+            i16_len::set_output_fixed_data(&SQL_OIC_CORE, info_value_ptr, string_length_ptr)
+        }
+        // SQL_SQL92_PREDICATES
+        InfoType::Sql92Predicates => {
+            // MongoSQL supports the following SQL-92 predicate operators.
+            const PREDICATES: u32 = SQL_SP_EXISTS
+                | SQL_SP_ISNOTNULL
+                | SQL_SP_ISNULL
+                | SQL_SP_LIKE
+                | SQL_SP_IN
+                | SQL_SP_BETWEEN
+                | SQL_SP_COMPARISON
+                | SQL_SP_QUANTIFIED_COMPARISON;
+            i16_len::set_output_fixed_data(&PREDICATES, info_value_ptr, string_length_ptr)
+        }
+        // SQL_SQL92_RELATIONAL_JOIN_OPERATORS
+        InfoType::Sql92RelationalJoinOperators => {
+            // MongoSQL supports the following SQL-92 JOIN operators.
+            const JOIN_OPS: u32 = SQL_SRJO_CROSS_JOIN
+                | SQL_SRJO_INNER_JOIN
+                | SQL_SRJO_LEFT_OUTER_JOIN
+                | SQL_SRJO_RIGHT_OUTER_JOIN;
+            i16_len::set_output_fixed_data(&JOIN_OPS, info_value_ptr, string_length_ptr)
+        }
+        // SQL_AGGREGATE_FUNCTIONS
+        InfoType::AggregateFunctions => {
+            // MongoSQL supports the following aggregate functions.
+            const AGG_FUNCTIONS: u32 = SQL_AF_AVG
+                | SQL_AF_COUNT
+                | SQL_AF_MAX
+                | SQL_AF_MIN
+                | SQL_AF_SUM
+                | SQL_AF_DISTINCT
+                | SQL_AF_ALL;
+            i16_len::set_output_fixed_data(&AGG_FUNCTIONS, info_value_ptr, string_length_ptr)
+        }
+        // SQL_RETURN_ESCAPE_CLAUSE
+        InfoType::ReturnEscapeClause => {
+            // MongoSQL has not support for the return escape clause since it is read-only.
+            i16_len::set_output_fixed_data(&SQL_U16_ZERO, info_value_ptr, string_length_ptr)
+        }
+        // SQL_CATALOG_NAME
+        InfoType::CatalogName => {
+            // MongoSQL does support catalog (database) names.
+            i16_len::set_output_wstring(
+                SQL_INFO_Y,
+                info_value_ptr as *mut WChar,
+                buffer_length as usize,
+                string_length_ptr,
+            )
+        }
+        // SQL_MAX_IDENTIFIER_LEN
+        InfoType::MaxIdentifierLen => {
+            // MongoSQL does not have a maximum identifier length.
```

Comment:
Debatable. We can't have an identifier longer than max document size :)

Reply:
Ha! True 😄